src/client.rs

```diff
@@ -0,0 +1,94 @@
+use crate::config::Config;
+
+use reqwest::blocking::RequestBuilder;
+use reqwest::Result;
+use reqwest::header;
+use serde::Deserialize;
+use std::process;
+
+const STATUS_URL: &str = "https://gw.lamarzocco.io/v1/home/machines/LM035401/status";
+const AUTH_URL: &str = "https://cms.lamarzocco.io/oauth/v2/token";
+
+#[derive(Deserialize)]
+struct StatusResponse {
+    data: Machine
+}
+
+#[derive(Deserialize)]
+struct Machine {
+    #[serde(rename = "MACHINE_STATUS")]
+    machine_status: String
+}
+
+pub fn get_status(cfg: Config) -> Result<String> {
+    let client = reqwest::blocking::Client::new();
+
+    println!("Fetching status...");
+    let resp = put_headers(cfg, client.get(STATUS_URL))
+        .send()?
+        .json::<StatusResponse>();
+
+    match resp {
+        Ok(status) => return Ok(status.data.machine_status),
+        Err(err) => return Err(err),
+    };
+}
+
+pub fn put_status(cfg: Config, on: bool) -> Result<String> {
+    let status = if on { "ON" } else { "STANDBY" };
+    let body = String::from(format!("{{\"status\":\"{}\"}}", status));
```

Comment:
You can use [raw strings](https://doc.rust-lang.org/reference/tokens.html#raw-string-literals) when dealing with JSON or anything that has a lot of noisy escapes:
```suggestion
    let body = String::from(format!(r#"{{"status":"{}"}}"#, status));
```